### PR TITLE
Allow users to manage archival status of owned pits

### PR DESCRIPTION
**Pull request by cyrus01337**
Solves #10

### CODE REVIEW

File: cogs/hideout/pits.py [hunk 1]
```diff
@@ -315,38 +322,43 @@ async def pit_archive(self, ctx: HideoutGuildContext, *, channel: discord.TextCh
...
+        elif pit is None or not isinstance(pit, discord.TextChannel):
+            raise commands.BadArgument('Could not find pit from id')
+        elif pits_category is None or not isinstance(pits_category, discord.CategoryChannel):
+            raise commands.BadArgument('Could not find valid pit category')
 
-        try:
-            pit: discord.TextChannel | None = ctx.guild.get_channel(record["pit_id"])  # type: ignore
-            if pit is None:
-                raise commands.BadArgument('Could not find pit from id')
+        archive_mode = ArchiveMode(record['archive_mode'])
```
Comment (LeoCx1000) [inline cogs/hideout/pits.py:344]: PostgreSQL enums return [strings](https://github.com/DuckBot-Discord/duck-hideout-manager-bot/blob/main/cogs/hideout/pits.py#L487) not integers, either check literals directly or update the enum values to be strings.

File: cogs/hideout/pits.py [hunk 2]
```diff
@@ -315,38 +322,43 @@ async def pit_archive(self, ctx: HideoutGuildContext, *, channel: discord.TextCh
...
+        is_not_counselor = ctx.guild.get_role(COUNSELORS_ROLE) not in ctx.author.roles
 
-            pits_category: Optional[discord.CategoryChannel] = ctx.guild.get_channel(PIT_CATEGORY)  # type: ignore
-            if pits_category is None:
-                raise commands.BadArgument('Could not find pit category')
+        if archive_mode is ArchiveMode.MANUAL and ctx.author != owner or is_not_counselor:
+            raise ActionNotExecutable('This pit was manually archived, only the pit owner and counsellors can unarchive it.')
+        elif archive_mode is ArchiveMode.INACTIVE and is_not_counselor:
+            raise ActionNotExecutable('This pit was marked as inactive, only the counsellors can unarchive it.')
```
Comment (LeoCx1000) [inline cogs/hideout/pits.py]: This logic should be the other way around, manual (archived by a counsellor) should not permit users from unarchiving it. Users should only be able to unarchive their own pit if it's been archived automatically.

```suggestion
        if archive_mode is ArchiveMode.INACTIVE and ctx.author != owner or is_not_counselor:
            raise ActionNotExecutable('This pit was marked as inactive, only the pit owner and counsellors can unarchive it.')
        elif archive_mode is ArchiveMode.MANUAL and is_not_counselor:
            raise ActionNotExecutable('This pit was manually archived, only the counsellors can unarchive it.')
```

**Comment by cyrus01337** [comment]
There were plans to include automatic archival, which will carry over in a separate issue and corresponding PR.

**Review by LeoCx1000 — CHANGES_REQUESTED** [review]
A few things I've noticed.

File: cogs/hideout/pits.py (hunk 1 above)
Comment (cyrus01337) [reply cogs/hideout/pits.py:344]: I have not dabbled with enums as much as I had liked, so this flew completely over me!

File: cogs/hideout/pits.py (hunk 2 above)
Comment (cyrus01337) [reply cogs/hideout/pits.py]: Swapped!

**Review by cyrus01337 — COMMENTED** [review]
Resolved.

**Review by LeoCx1000 — APPROVED** [review]

**Pull request merged**